tools/chef/cookbooks/px_dev_extras/recipes/packaging.rb

```diff
@@ -0,0 +1,27 @@
+# Copyright 2018- The Pixie Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Packaging utilities only need to run on Linux
+if node.platform_family?('debian')
+  apt_package ['ruby-dev', 'rpm'] do
+    action :upgrade
+  end
+
+  execute 'install_fpm' do
+    command '/usr/bin/gem install fpm'
+    action :run
+  end
```

Comment:
I retested using `gem_package` and it installs fpm within `/opt/chef/embedded` as I saw during my previous test. Below is the `opensnoop` output filtered by `fpm` files and the `git diff` showing what I tested:

```
$ git diff
diff --git a/tools/chef/cookbooks/px_dev_extras/recipes/packaging.rb b/tools/chef/cookbooks/px_dev_extras/recipes/packaging.rb
index 7d9177935..2292ec4d6 100644
--- a/tools/chef/cookbooks/px_dev_extras/recipes/packaging.rb
+++ b/tools/chef/cookbooks/px_dev_extras/recipes/packaging.rb
@@ -20,8 +20,11 @@ if node.platform_family?('debian')
     action :upgrade
   end

-  execute 'install_fpm' do
-    command '/usr/bin/gem install fpm'
-    action :run
+  gem_package 'fpm' do
+    gem_binary '/usr/bin/gem'
   end
+  # execute 'install_fpm' do
+  #   command '/usr/bin/gem install fpm'
+  #   action :run
+  # end
 end


$ grep fpm chef_opensnoop.txt | grep -v 'git\|php'
1759223 python3             5   0 /tmp/pip-build-tracker-dioyjfpm
1786655 chef-solo           9   0 /root/.local/share/gem/specs/index.rubygems.org%443/quick/Marshal.4.8/fpm-1.15.1.gemspec
1787105 ruby                6   0 /root/.local/share/gem/specs/index.rubygems.org%443/quick/Marshal.4.8/fpm-1.15.1.gemspec
1787105 request_set.rb*     6   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/cache/fpm-1.15.1.gem
1787105 ruby                6   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/cache/fpm-1.15.1.gem
1787105 ruby                6   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/cache/fpm-1.15.1.gem
1787105 ruby                6   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/cache/fpm-1.15.1.gem
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/CHANGELOG.rst
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/CONTRIBUTORS
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/LICENSE
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/bin/fpm
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/command.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/errors.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/namespace.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/apk.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/cpan.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/deb.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/dir.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/empty.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/freebsd.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/gem.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/npm.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/osxpkg.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/p5p.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/pacman.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/pear.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/pkgin.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/pleaserun.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/puppet.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/pyfpm/__init__.py
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/pyfpm/get_metadata.py
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/python.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/rpm.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/sh.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/snap.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/solaris.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/tar.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/virtualenv.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/package/zip.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/rake_task.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/util.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/util/tar_writer.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/lib/fpm/version.rb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/deb.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/deb/changelog.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/deb/deb.changes.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/deb/ldconfig.sh.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/deb/postinst_upgrade.sh.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/deb/postrm_upgrade.sh.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/deb/preinst_upgrade.sh.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/deb/prerm_upgrade.sh.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/osxpkg.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/p5p_metadata.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/pacman.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/pacman/INSTALL.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/pleaserun/generate-cleanup.sh
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/pleaserun/install-path.sh
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/pleaserun/install.sh
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/pleaserun/scripts/after-install.sh
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/pleaserun/scripts/before-remove.sh
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/puppet/package.pp.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/puppet/package/remove.pp.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/rpm.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/rpm/filesystem_list
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/sh.erb
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/templates/solaris.erb
1787105 ruby                6   0 /opt/chef/embedded/bin/fpm
1787105 ruby                7   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/gems/fpm-1.15.1/bin/fpm
1787105 ruby                6   0 /opt/chef/embedded/lib/ruby/gems/3.1.0/specifications/fpm-1.15.1.gemspec
1788739 rm                  4   0 fpm-1.15.1
1788739 rm                  4   0 fpm
```